### PR TITLE
Migrations for unified accounts switch

### PR DESCRIPTION
**Pull request by dastanfv**
migrates `sudo` key to a controlled `evm` key

### CODE REVIEW

File: ownership-chain/runtime/src/migrations/v1.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		}
+
+		/// Migrate sudo key.
+		///
+		/// Simply remove old sudo key and insert new sudo key.
+		///
+		/// This function is called during the runtime upgrade process.
+		///
+		/// NOTE: storage version of `pallet_sudo` can not be updated because it's an external
+		/// pallet, i.e we have to set `#[pallet::storage_version(VERSION)]` attribute.
```
Comment (magecnion) [inline ownership-chain/runtime/src/migrations/v1.rs:49]: how do we ensure that migration only runs once then?

File: ownership-chain/runtime/src/migrations/v1.rs [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+		fn on_runtime_upgrade() -> frame_support::weights::Weight {
+			let raw_sudo_key = storage_prefix(
+				<pallet_sudo::Pallet<T> as PalletInfoAccess>::name().as_bytes(),
+				b"Key",
+			);
+
+			let new_sudo: [u8; 20] =
+				hex_literal::hex!("47A4320be4B65BF73112E068dc637883490F5b04").into();
```
Comment (magecnion) [inline ownership-chain/runtime/src/migrations/v1.rs:57]: do we need that this account was prefunded with some tokens?

File: ownership-chain/runtime/src/migrations/v1.rs (hunk 1 above)
Comment (dastanfv) [reply ownership-chain/runtime/src/migrations/v1.rs:49]: we remove it once it's run. i.e next runtime upgrade we don't run anything from `v1.rs`

File: ownership-chain/runtime/src/migrations/v1.rs (hunk 2 above)
Comment (dastanfv) [reply ownership-chain/runtime/src/migrations/v1.rs:57]: not necessarily, we have `pallet_balances::set_balance` sudo call that can set the balance. but I guess, it should have some balance for transactions, I will mint it here in this migration.

Comment (dastanfv) [reply ownership-chain/runtime/src/migrations/v1.rs:57]: addressed it

**Review by magecnion — APPROVED** [review]

**Pull request merged**